src/boundaries.rs

```diff
@@ -102,7 +102,55 @@ pub fn build_boundary(
     relation: &osmpbfreader::Relation,
     objects: &BTreeMap<osmpbfreader::OsmId, osmpbfreader::OsmObj>,
 ) -> Option<MultiPolygon<f64>> {
-    let roles = ["outer", "enclave", ""];
+    use geo::algorithm::contains::Contains;
+
+    let outer_polys = build_boundary_parts(relation, objects, vec!["outer", "enclave", ""]);
+    let inner_polys = build_boundary_parts(relation, objects, vec!["inner"]);
+    let mut multipoly = MultiPolygon(vec![]);
+
+    if inner_polys.is_none() {
+        return outer_polys;
+    }
+    let mut inner_polys = inner_polys.unwrap();
+
+    if outer_polys.is_some() {
+        let outer_polys = outer_polys.unwrap();
+        for outer in outer_polys.0 {
```

Comment:
This algorithm does not handle the "donut-inside-donut" situation.  
A solution might be to sort `outer_polys` and `inner_polys` by area first ?